modules/javafx.controls/src/main/java/javafx/scene/control/skin/VirtualFlow.java

```diff
@@ -2592,12 +2592,14 @@ private void updateScrollBarsAndCells(boolean recreate) {
         // thumb can be appropriately sized
         if (recreate && (lengthBar.isVisible() || Properties.IS_TOUCH_SUPPORTED)) {
             final int cellCount = getCellCount();
-            int numCellsVisibleOnScreen = 0;
+            float numCellsVisibleOnScreen = 0;
             for (int i = 0, max = cells.size(); i < max; i++) {
                 T cell = cells.get(i);
                 if (cell != null && !cell.isEmpty()) {
-                    sumCellLength += (isVertical ? cell.getHeight() : cell.getWidth());
+                    double lengthToAdd = (isVertical ? cell.getHeight() : cell.getWidth());
+                    sumCellLength += lengthToAdd;
                     if (sumCellLength > flowLength) {
+                        numCellsVisibleOnScreen += ( 1 - (sumCellLength - flowLength )/ lengthToAdd);
```

Comment:
Minor: remove the spaces after the `(` and before the `)`, and add a space before the `/` like this:

```
                        numCellsVisibleOnScreen += (1 - (sumCellLength - flowLength ) / lengthToAdd);
```